pkg/kubelet/kubelet.go

```diff
@@ -2964,8 +2980,17 @@ func (kl *Kubelet) handlePodResourcesResize(pod *v1.Pod, podStatus *kubecontaine
 			return
 		}
 		if kl.isPodResizeInProgress(allocatedPod, podStatus) {
-			// If a resize is in progress, make sure the cache has the correct state in case the Kubelet restarted.
-			kl.statusManager.SetPodResizeInProgressCondition(pod.UID, "", "", false)
+			// If a resize is in progress, make sure the cache has the correct state.
+			if len(kl.statusManager.GetPodResizeConditions(pod.UID)) == 2 {
+				// The new pod resize is either deferred or infeasible, which means if there is a pod
+				// resize in progress, it is associated with an older resize. In this case, the
+				// condition reflecting the older resize should also have the older observedGeneration.
+				gen := kl.statusManager.GetPodResizeInProgressCondition(pod.UID).ObservedGeneration
+				kl.statusManager.SetPodResizeInProgressCondition(pod.UID, "", "", gen, false)
+			} else {
+				kl.statusManager.SetPodResizeInProgressCondition(pod.UID, "", "", pod.Generation, false)
```

Comment:
You could hit this case if there's a resize in progress, but no new resize request (the `!updated` case below). I think in that case you just want to leave the condition as is?

I guess it doesn't really matter if the generation was bumped for a non-resize change, but it's a little weird.